internal/nearcache/repairing_task.go

```diff
@@ -0,0 +1,568 @@
+	mcs := make([]*MetaDataContainer, partitionCount)
+	for i := int32(0); i < partitionCount; i++ {
+		mcs[i] = NewMetaDataContainer()
+	}
+	sk := nc.Config().SerializeKeys
+	f := func(key interface{}) (interface{}, error) {
+		if sk {
+			return key, nil
+		}
+		return ss.ToObject(key.(serialization.Data))
+	}
+	return RepairingHandler{
+		name:               name,
+		nc:                 nc,
+		metadataContainers: mcs,
+		toNearCacheKey:     f,
+		ss:                 ss,
+		ps:                 ps,
+		lg:                 lg,
+	}
+}
+
+func (h RepairingHandler) Name() string {
+	return h.name
+}
+
+func (h RepairingHandler) GetMetaDataContainer(partition int32) *MetaDataContainer {
+	return h.metadataContainers[partition]
+}
+
+func (h *RepairingHandler) UpdateLastKnownStaleSequence(md *MetaDataContainer, partition int32) {
+	var lastReceived int64
+	var lastKnown int64
+	for {
+		lastReceived = md.Sequence()
+		lastKnown = md.StaleSequence()
+		if lastKnown >= lastReceived {
+			break
+		}
+		if md.CASStaleSequence(lastKnown, lastReceived) {
+			break
+		}
+	}
+	h.lg.Trace(func() string {
+		return fmt.Sprintf("stale sequences updated:[map=%s,partition=%d,lowerSequencesStaleThan=%d,lastReceivedSequence=%d]",
+			h.name, partition, md.StaleSequence(), md.Sequence())
+	})
+}
+
+// Handle handles a single invalidation.
+func (h *RepairingHandler) Handle(key serialization.Data, source, partition types.UUID, seq int64) error {
+	// port of: com.hazelcast.internal.nearcache.impl.invalidation.RepairingHandler#handle(com.hazelcast.internal.serialization.Data, java.util.UUID, java.util.UUID, long)
+	// Apply invalidation if it's not originated by local member/client.
+	// Since local Near Caches are invalidated immediately there is no need to invalidate them twice.
+	if source != partition {
+		if key == nil {
+			h.nc.Clear()
+		} else {
+			k, err := h.toNearCacheKey(key)
+			if err != nil {
+				return err
+			}
+			h.nc.Invalidate(k)
+		}
+	}
+	pid, err := h.getPartitionIDOrDefault(key)
+	if err != nil {
+		return err
+	}
+	h.CheckOrRepairUUID(pid, partition)
+	h.CheckOrRepairSequence(pid, seq, false)
+	return nil
+}
+
+// HandleBatch handles a batch of validations.
+func (h *RepairingHandler) HandleBatch(keys []serialization.Data, sources []types.UUID, partitions []types.UUID, seqs []int64) error {
+	// port of: com.hazelcast.internal.nearcache.impl.invalidation.RepairingHandler#handle(java.util.Collection<com.hazelcast.internal.serialization.Data>, java.util.Collection<java.util.UUID>, java.util.Collection<java.util.UUID>, java.util.Collection<java.lang.Long>)
+	// assumes len(keys) == len(source) -- len(partitions) == len(seqs)
+	sz := len(keys)
+	for i := 0; i < sz; i++ {
+		if err := h.Handle(keys[i], sources[i], partitions[i], seqs[i]); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (h *RepairingHandler) CheckOrRepairUUID(partition int32, new types.UUID) {
+	// this method may be called concurrently: anti-entropy, event service
```

Comment:
```suggestion
	// this method may be called concurrently by anti-entropy and event service
```

Reply:
I will keep the original one if you don't have a strong opinion about it.

Reply:
Well, of course I have no strong opinion, but at first read I could not understand what it meant until I read Java side. Hence I have commented